libhb/muxavformat.c

```diff
@@ -56,6 +56,29 @@ struct hb_mux_object_s
     hb_mux_data_t    ** tracks;
 };
 
+static hb_rational_t vrates[] =
```

Comment:
How did you determine that these are the rates that can be used? I suggest a comment that says where these values came from (mp4 spec?) and maybe rename vrates => mp4_vrates since they appear to be mp4 specific?

Reply:
Those are same fractions as the ones in common.c hb_video_rates, but simplified. Somewhere along the way hb_reduce() is called on the vrate so they can be easily compared.
Probably it's better to call hb_reduce() on the vrate just to be sure, I see now that there is some code at line 660 that sets the track avg_frame_rate, I'll try to move the time_base part there too.

Reply:
So it's a much truncated version of hb_video_rates? Any reason for truncating to just those particular rates? Maybe add the function in common.c to allow all supported rates?

Aside, hb_video_rates and hb_rate_t in general are kind of a train wreck in that the numerator isn't explicitly given or even commented upon :stuck_out_tongue_winking_eye: hb_rate_t usage is pretty limited, so maybe worth cleaning that up at some point by adding the numerator to the struct.

Reply:
The others rate can be represented in the default 90000 timescale without any jitter, so there is no need to set a custom timescale.
Now that I think about I can just check if the vrate can be converted to 90000 without any remainder, so the additional vrates array is not needed.

The phantom 27000000 numerator 😆, there is a define but not in a public header, the MacGui simply hardcodes it when needed ahah.

Reply:
> I can just check if the vrate can be converted to 90000 without any remainder

Simpler and almost self-explanatory. I like it!